src/avram/queryable.cr

```diff
@@ -196,7 +196,10 @@ module Avram::Queryable(T)
   end
 
   def select_count : Int64
-    exec_scalar(&.select_count).as(Int64)
+    table = "(#{query.statement}) AS temp"
+    new_query = Avram::QueryBuilder.new(table).select_count
```

Comment:
This could be cleared up if we implemented https://github.com/luckyframework/avram/issues/682